utilize oc adm release extract --included options

jianlinliu opened the pull request:
1. utilize oc adm release extract --included options to improve CR manifest generation
2. introduce `EXTRACT_MANIFEST_INCLUDED` to control the behavior on different versions of job config
3. add new jobs `aws-ipi-disconnected-sts-basecap-none-addcap-f28` to match designed cases for both fresh install and upgrade scenarios
4. set `EXTRACT_MANIFEST_INCLUDED: "false"` on 4.14 ec jobs, once stable build come out, will update them in another PR.
5. reset `PATH` to get injected cli when the step is using upi-installer:tag image

jiajliu commented on ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-stable-4.13.yaml:
```diff
@@ -193,6 +193,22 @@ tests:
     - ref: cucushift-installer-check
     - chain: openshift-upgrade-qe-test-disconnected
     workflow: cucushift-installer-rehearse-aws-ipi-disconnected-private-sno
+- as: aws-ipi-disconnected-sts-basecap-none-addcap-f28
```
We need add extra pre-steps before upgrade for this profile in following pr, right?

jianlinliu replied:
I think that depends on if there is some different CRs are introduced between 4.13 and 4.14. If there is no difference by chance, then maybe do not need a pre-upgrade step for the dedicated CR operation. cc @jianping-shu to confirm.

jianlinliu replied:
But for a long-term gold, we should have the pre-upgrade step to check the CR difference between versions, so that ensure the required CRs is provided before upgrade.

jiajliu replied:
With `--included` option, we can run the cmd before upgrade to check the update, and that's why we add one more upgrade case for it. 
My concern is that this new added upgrade job will fail until we add step6&step7 in the [case](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-66839). @jianping-shu pls correct me if my understanding is wrong. 
As for the integration for step6&7, I'm still thinking about the best way for the case automation, just as suggested in automation meeting just now.

jianping-shu replied:
Let me double check the 4.13 CRs vs. 4.14 ones and update later

jianlinliu replied:
Take an example, if I understand right, source cluster has 5 CRs, target cluster requires 6 CRs, once that, before upgrade, we definitely need to create one more that is missing in the source cluster whatever from the regular team ci view or the OTA automation view.

jiajliu replied:
Yeah, if we definitely need one more step for upgrade here, how about updating `f28` to `f365` before we support it actually? and re-update it back once integrating the step in ci. Or else, the upgrade failure will be caught by the alert and we need do failure analysis:)

jianping-shu replied:
That's right!
I just double checked the 4.14 aws CRs changes vs. 4.13.
There are permission added for 2 CRs i.e. cluster-ingress-operator and cluster-storage-operator in 4.14. You can find the details in Column K in [CredentialsRequest tracking](https://docs.google.com/spreadsheets/d/1K4lZpA9hp9uW5XpieUJeqt8-K8Ff5fVPcFyH6W943Uk/edit#gid=0)
According to past experiences, the upgrade might be successful without pre-steps unless the added permission is mandatory for the operator startup.

As to upgradeable-to annotation, its logic should be already done in function cco_annotation for upgrade.

jiajliu replied:
thanks for the confirm, then we could have a test on the upgrade once a green build available.

jianlinliu replied:
agree, we can have a test to see the results.

jiajliu commented on ci-operator/step-registry/aws/provision/cco-manual-users/static/aws-provision-cco-manual-users-static-ref.yaml:
```diff
@@ -18,6 +18,10 @@ ref:
+  - name: EXTRACT_MANIFEST_INCLUDED
+    default: "false"
+    documentation: |-
+      Determine whether to exclude manifests that are not expected to be included in the cluster when extracti
```
typo `extracti`

jianlinliu replied:
done

jianlinliu commented:
Once the oc new feature landed into a green nightly build, I will trigger some jobs to validate the chagne.

yunjiang29 commented:
@jianlinliu the CR process step should happen after `ipi-conf` where  TechPreview and CapabilitySet have been configured, I found some chains need to be updated:

`- ref: aws-provision-cco-manual-users-static`
```
cucushift-installer-rehearse-aws-ipi-disconnected-private-network-mtu-localzone-provision
cucushift-installer-rehearse-aws-ipi-disconnected-private-provision
cucushift-installer-rehearse-aws-ipi-disconnected-private-sdn-provision
cucushift-installer-rehearse-aws-ipi-disconnected-private-sno-provision
cucushift-installer-rehearse-aws-ipi-disconnected-private-techpreview-provision
quay-tests-aws-ipi-disconnected-private-provision

```

jianlinliu commented:
>  the CR process step should happen after ipi-conf where TechPreview and CapabilitySet have been configured, I found some chains need to be updated

Good catch, fixed them. Beside them, I also updated `aws-c2s-init-token-service` together.

jianlinliu commented:
/pj-rehearse

jianlinliu commented:
/pj-rehearse

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
In [periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-addcap-f28](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/42625/rehearse-42625-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-disconnected-sts-basecap-none-addcap-f28/1696728351378509824/artifacts/aws-ipi-disconnected-sts-basecap-none-addcap-f28/ipi-conf-aws-oidc-creds-provision/build-log.txt):
```
CR manifest files:
0000_30_machine-api-operator_00_credentials-request.yaml
0000_50_cloud-credential-operator_05-iam-ro-credentialsrequest.yaml
0000_50_cluster-image-registry-operator_01-registry-credentials-request.yaml
0000_50_cluster-ingress-operator_00-ingress-credentials-request.yaml
0000_50_cluster-network-operator_02-cncc-credentials.yaml
```

jianlinliu commented:
/pj-rehearse max

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sdn-rhcos-rhel8-f28-destructive

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sdn-rhcos-rhel8-f28-destructive

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sdn-rhcos-rhel8-f28-destructive

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sdn-rhcos-rhel8-f28-destructive

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-stack-ipi-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-nutanix-ipi-compact-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-disconnected-sts-basecap-none-addcap-f28

jianlinliu commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-f14

jianlinliu commented:
@jiajliu @jianping-shu [aws-ipi-disconnected-sts-basecap-none-addcap-f28](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/42625/rehearse-42625-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-disconnected-sts-basecap-none-addcap-f28/1697411761533095936/artifacts/aws-ipi-disconnected-sts-basecap-none-addcap-f28/cucushift-upgrade-toimage/build-log.txt) upgrade job get passed.

jianlinliu commented:
@jinyunma @yunjiang29 @dongboyan77 @LiZhang19817 @liangxia  The PR get ready to merge, pls review.

jianlinliu commented:
/pj-rehearse ack

yunjiang29 commented:
/lgtm

liangxia commented:
/lgtm

jinyunma commented:
/lgtm

dongboyan77 commented:
/lgtm

liangxia commented:
/pj-rehearse abort
